packages/ember-glimmer/tests/integration/helpers/element-action-test.js

```diff
@@ -0,0 +1,1340 @@
+      event[prop] = value;
+
+      component.$('button').trigger(event);
+
+      if (expected) {
+        assert.ok(showCalled, `should call action with ${prop}:${value}`);
+        assert.ok(event.isDefaultPrevented(), 'should prevent default');
+      } else {
+        assert.notOk(showCalled, `should not call action with ${prop}:${value}`);
+        assert.notOk(event.isDefaultPrevented(), 'should not prevent default');
+      }
+    }
+
+    checkClick('ctrlKey', true, false);
+    checkClick('altKey', true, false);
+    checkClick('metaKey', true, false);
+    checkClick('shiftKey', true, false);
+    checkClick('which', 2, false);
+
+    checkClick('which', 1, true);
+    checkClick('which', undefined, true); // IE <9
+  }
+
+  ['@test it can trigger actions for keyboard events']() {
+    let showCalled = false;
+
+    let ExampleComponent = Component.extend({
+      actions: {
+        show() {
+          showCalled = true;
+        }
+      }
+    });
+
+    this.registerComponent('example-component', {
+      ComponentClass: ExampleComponent,
+      template: '<input type="text" {{action "show" on="keyUp"}}>'
+    });
+
+    this.render('{{example-component}}');
+
+    this.runTask(() => {
+      let event = jQuery.Event('keyup');
+      event.char = 'a';
+      event.which = 65;
+      this.$('input').trigger(event);
+    });
+
+    this.assert.ok(showCalled, 'the action was called with keyup');
+  }
+
+  ['@test a quoteless parameter should allow dynamic lookup of the actionName']() {
+    let lastAction;
+    let actionOrder = [];
+    let component;
+
+    let ExampleComponent = Component.extend({
+      init() {
+        this._super(...arguments);
+        component = this;
+      },
+      hookMeUp: 'rock',
+      actions: {
+        rock() {
+          lastAction = 'rock';
+          actionOrder.push('rock');
+        },
+        paper() {
+          lastAction = 'paper';
+          actionOrder.push('paper');
+        },
+        scissors() {
+          lastAction = 'scissors';
+          actionOrder.push('scissors');
+        }
+      }
+    });
+
+    this.registerComponent('example-component', {
+      ComponentClass: ExampleComponent,
+      template: '<a id="bound-param" {{action hookMeUp}}>Whistle tips go woop woooop</a>'
+    });
+
+    this.render('{{example-component}}');
+
+    let test = this;
+
+    function testBoundAction(propertyValue) {
+      test.runTask(() => {
+        component.set('hookMeUp', propertyValue);
+      });
+
+      test.runTask(() => {
+        component.$('#bound-param').click();
+      });
+
+      test.assert.ok(lastAction, propertyValue, `lastAction set to ${propertyValue}`);
+    }
+
+    testBoundAction('rock');
+    testBoundAction('paper');
+    testBoundAction('scissors');
+
+    this.assert.deepEqual(actionOrder, ['rock', 'paper', 'scissors'], 'action name was looked up properly');
+  }
+
+  ['@test a quoteless parameter should lookup actionName in context [DEPRECATED]']() {
```

Comment:
It doesn't seem like it. The original test name included `[DEPRECATED]` so I carried it along.